draft-ppm-protocol.md

```diff
@@ -415,69 +405,58 @@ PPM has three major interactions which need to be defined:
+We start with some basic type definitions used in other messages.
 
 ~~~
-struct {
-  opaque uuid[16];
-  Url aggregator_endpoints<1..2^16-1>;
-  HpkeConfig collector_config;
-  uint64 max_batch_lifetime;
-  uint64 min_batch_size;
-  Duration min_batch_duration;
-  Proto proto;
-  uint16 length; // Length of the remainder.
-  select (Param.proto) {
-    case prio: PrioParam;
-    case hits: HitsParam;
-  }
-} Param;
-
 enum { prio(0), hits(1) } Proto;
 
+ /* ASCII encoded URL. e.g., "https://example.com" */
 opaque Url<1..2^16-1>;
 
 Duration uint64; /* Number of seconds elapsed between two instants */
 
 Time uint64; /* seconds elapsed since start of UNIX epoch */
+
+/* An interval of time, where start is included and end is excluded */
+struct {
+  Time start;
+  Time end;
+} Interval;
 ~~~
 
-* `uuid`: A unique sequence of bytes used to ensure that two otherwise
-  identical `Param` instances will have distinct `TaskID`s. It is
-  RECOMMENDED that this be set to a random 16-byte string derived from a
-  cryptographically secure pseudorandom number generator.
-* `aggregator_endpoints`: Vector of URLs relative to which an aggregator's API
-  endpoints can be found. The leader's endpoint MUST be the first in the vector.
-  The order of the `encrypted_input_shares` in a `Report` (see
-  {{uploading-reports}}) MUST be the same as the order in which aggregators
-  appear in this vector.
-* `collector_config`: The HPKE configuration of the collector (described in
-  {{key-config}}). Putting the collector's HPKE configuration directly in
-  `struct PPMParam` absolves collectors of the burden of operating an HTTP
-  server. See [#102](https://github.com/abetterinternet/ppm-specification/issues/102)
-  for discussion.
-* `max_batch_lifetime`: The maximum number of times a batch of reports may be
-  used in a collect request.
-* `min_batch_size`: The minimum batch size, i.e., the minimum number of reports
-  that appear in a batch.
-* `min_batch_duration`: The minimum batch duration, i.e., the minimum time
-  difference between the oldest and newest report in a batch. This defines the
-  boundaries with which the batch interval of each collect request must be
-  aligned. (See {{batch-parameter-validation}}.)
-* `proto`: The PPM protocol, e.g., Prio or Hits. The rest of the structure
-  contains the protocol specific parameters.
-
-The *task ID* is derived from the PPM parameters as:
+## Task Configuration {#task-configuration}
+
+Prior to the start of the protocol, each endpoint in the system must agree on
+the configuration for each task. A task is uniquely identified by its task ID:
 
 ~~~
-task_id = SHA-256(param)
+opaque TaskId[32];
 ~~~
 
-Where param is the serialization of Param.
-
+A `TaskId` is a globally unique sequence of bytes. It is RECOMMENDED that this
+be set to a random string output by a cryptographically secure pseudorandom
+number generator. Each task has the following parameters associated with it:
+
```

Comment:
Bullets describing parameters with names used elsewhere in the doc should begin with the name. E.g.,:
> * `max_batch_lifetime` is the ...